src/GerdaFitter.cc

```diff
@@ -259,8 +260,21 @@ GerdaFitter::GerdaFitter(json outconfig) : config(outconfig) {
 
         // loop over requested histograms in file
         for (auto& elh : el.value().items()) {
+            // get rebin factor before reading all histograms
+            auto rebin_x = elh.value().value("rebin-factor-x", 1) * elh.value().value("rebin-factor", 1);
+            auto rebin_y = elh.value().value("rebin-factor-y", 1);
+            if (rebin_x != 1)
+                BCLog::OutDetail("using rebin X factor = " + std::to_string(rebin_x) + " for dataset '" + elh.key() + "'");
+            if (rebin_y != 1)
+                BCLog::OutDetail("using rebin Y factor = " + std::to_string(rebin_y) + " for dataset '" + elh.key() + "'");
+
             BCLog::OutDebug("getting data histogram '" + elh.key() + "' from " + el.key());
             auto th = dynamic_cast<TH1*>(_tf.Get(elh.key().c_str()));
+            // rebin if requested
+            th->Rebin(rebin_x);
```

Comment:
Ah sorry, this is what you do here? but then the Y rebin gets applied on top of this?

Reply:
Rebin is always just in x. Rbiny is applied later in the code.